src/ngraph_builder1.cc

```diff
@@ -0,0 +1,389 @@
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_builder1.h"
+#include "ngraph_translateops.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+Status Builder1::TranslateGraph(
+    const std::vector<TensorShape>& inputs,
+    const std::vector<const Tensor*>& static_input_map,
+    shared_ptr<ng::Function>& ng_function) {
+  TF_RETURN_IF_ERROR(Initialize());
+
+  vector<shared_ptr<ng::op::Parameter>> ng_parameter_list;
+  TF_RETURN_IF_ERROR(GetInputParams(inputs, tf_params, ng_parameter_list));
+
+  TF_RETURN_IF_ERROR(TranslateEachOp(tf_ops, static_input_map));
+
+  VectNg ng_result_list;
+  TF_RETURN_IF_ERROR(GetOutputNodes(tf_ret_vals, ng_result_list));
+
+  // Create the nGraph function.
+  try {
+    ng_function = make_shared<ng::Function>(ng_result_list, ng_parameter_list);
+  } catch (...) {
+    return errors::Unimplemented("Unable to create nGraph function");
+  }
+
+  //
+  // Request row-major layout on results.
+  //
+  for (auto result : ng_function->get_results()) {
+    result->set_needs_default_layout(true);
+  }
+  return Status::OK();
+}
+
+Status Builder1::TranslateGraph(
+    OpKernelContext* ctx, std::shared_ptr<ngraph::Function>& ng_function) {
+  TF_RETURN_IF_ERROR(Initialize());
+
+  std::vector<const Tensor*> static_input_map;
+
+  std::vector<TensorShape> inputs(ctx->num_inputs());
+
+  static_input_map.resize(ctx->num_inputs());
+  for (int i = 0; i < ctx->num_inputs(); i++) {
+    const Tensor& input_tensor = ctx->input(i);
+    if (m_input_is_static[i]) {
+      static_input_map[i] = &input_tensor;
+    }
+    inputs[i] = input_tensor.shape();
+  }
+
+  return TranslateGraph(inputs, static_input_map, ng_function);
+}
+
+Status Builder1::TranslateEachOp(
+    const vector<const Node*>& tf_ops,
+    const std::vector<const Tensor*>& static_input_map) {
+  // Create the nGraph ops from TensorFlow ops.
+  for (auto op : tf_ops) {
+    NGRAPH_VLOG(2) << "Constructing op " << op->name() << " which is "
+                   << op->type_string();
+
+    Builder1::TranslatorFn translate_fn;
+    vector<int> input_indexes;
+    TF_RETURN_IF_ERROR(
+        GetOpTranslationRequirements(op, translate_fn, input_indexes));
+    // input_indexes can be size 0 (to indicate/handle variadic inputs nodes
+    // like Addn)
+    VectNg subgraph_out_nodes(op->num_outputs());
+
+    bool variadic_input = input_indexes.size() == 0;
+    int num_inputs = variadic_input ? op->num_inputs() : input_indexes.size();
+    VectNg ng_arg_vec(num_inputs);
+    if (op->type_string() != "Const") {
+      for (int idx = 0; idx < num_inputs; idx++) {
+        TF_RETURN_IF_ERROR(GetInputNode(
+            op, (variadic_input ? idx : input_indexes[idx]), &ng_arg_vec[idx]));
+      }
+    }
+    TF_RETURN_IF_ERROR(
+        translate_fn(op, ng_arg_vec, static_input_map, subgraph_out_nodes));
```

Comment:
I wonder if it would be possible to extract and pass just the static input tensors that are actually being used by this node, by examining `_ngraph_static_inputs` (or whatever we called it) here?

Reply:
Try feeling this idea out as you implement some ops which have static inputs (like, say, convolution backprop)

Reply:
the padding issue: I ran the math on both implementations, they *seem* to be the same. But who knows what crazy corner case might show up. In case the current implementation works, lets keep that